ballista/scheduler/src/state/task_manager.rs

```diff
@@ -405,19 +478,19 @@ impl<T: 'static + AsLogicalPlan, U: 'static + AsExecutionPlan> TaskManager<T, U>
             ]
         };
 
-        let _res = if let Some(graph) = self.remove_active_execution_graph(job_id).await {
+        if let Some(graph) = self.remove_active_execution_graph(job_id).await {
             let mut graph = graph.write().await;
             let previous_status = graph.status();
             graph.fail_job(failure_reason);
-            let value = self.encode_execution_graph(graph.clone())?;
+
+            let value = encode_protobuf(&graph.status())?;
```

Comment:
First bug uncovered while testing the event loop proper :) When a job would fail we would save the entire graph into `FailedJobs` here but in all other scenarios we only save the status